### ISSUE TITLE
Merge downstream buildsystem

### ISSUE THREAD

**Issue by p01arst0rm** [luceneplusplus/LucenePlusPlus#139]
Not sure of the best way to contact project mods, so posting here. 
There is a downstream fork of LucenePlusPlus that replaces the cmake
build system with a meson one, which makes building the source far
easy on non-linux systems. It also has a couple source tweaks to improve 
readability. 

Would you be interested in merging it once the port is complete?
https://github.com/hatchet-player/LucenePlusPlus/tree/meson

**Comment by alanw** [comment]
Interesting.

If possible, I'd like to keep CMake because so many build systems/distributions depend on it. Is it possible to support both?

**Comment by p01arst0rm** [comment]
> Interesting.
> 
> If possible, I'd like to keep CMake because so many build systems/distributions depend on it. Is it possible to support both?

yep; its perfectly possible to have both buildsystems in the build directory.

**Comment by p01arst0rm** [comment]
if you want an example of a dual build system project, look up mesa3d/waffle

**Comment by p01arst0rm** [comment]
@alanw the fork now contains a completely rebuilt cmake buildsystem prototype (missing doxygen & config files atm). If youre intreseted, ill finish it up :)

**Comment by p01arst0rm** [comment]
got the tester working in windows now... :)

![image](https://user-images.githubusercontent.com/11366022/90508914-6799fe00-e150-11ea-81c6-3151a730ff4d.png)
